### PR TITLE
suffix workspace name when creating from devfile and name already exists

### PR DESCRIPTION
**Pull request by sparkoo**
Signed-off-by: Michal Vala <mvala@redhat.com>

### What does this PR do?
When creating a workspace from devfile, it now tries to suffix workspace name with `_[1-99]` before failing.
I've decided to fix it in `Manager` layer. `DAO` is too low, should be keep simple and would affect also creation from `WorkspaceConfig`. Alternatively might be fixed in `Service` layer, but there it might lead to duplicates when we will need to call manager from multiple places (Service is only call in the current state) and I think it's responsibility of `Manager`.

### What issues does this PR fix or reference?
#13794 #13683 

#### Release Notes
n/a

#### Docs PR
n/a

### CODE REVIEW

**Comment by sparkoo** [comment]
ci-test

**Comment by musienko-maxim** [comment]
Can one of the admins verify this PR?

**Comment by musienko-maxim** [comment]
E2E Happy path tests of Eclipse Che Single User on K8S (minikube v1.1.1) has been successful:
- [Build details](https://codeready-workspaces-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/CHE-7-Maven-PR-build-checker-codenvy-slave9/41/)
- docker image: docker.io/maxura/che-server:14129

**Review by mshaposhnik — APPROVED** [review]

**Comment by skabashnyuk** [comment]
let's discuss more the generated name format. The current implementation is too heavy for backend

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java [hunk 1]
```diff
@@ -593,4 +606,43 @@ private WorkspaceImpl getByKey(String key) throws NotFoundException, ServerExcep
...
+        getWorkspace(candidateName, namespace);
+        candidateName = baseName + WORKSPACE_NAME_CONFLICT_SUFFIX_SEPARATOR + counter;
+        counter++;
+      } while (counter < WORKSPACE_NAME_CONFLICT_SUFFIX_MAX);
+      throw new ConflictException(
+          format(
+              "Workspace with name '%s' including '_[1-%d]' in namespace '%s' already exists.",
+              baseName, WORKSPACE_NAME_CONFLICT_SUFFIX_MAX, namespace));
+    } catch (NotFoundException nfe) {
+      newMetadata.setName(candidateName);
```
[the remark this answers is not in the document]
Comment (sparkoo) [reply wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java:643]: thanks, I'll think about that in new proposed solution.

**Comment by sparkoo** [comment]
closing this one in favour of this proposal https://github.com/eclipse/che/issues/13794#issuecomment-518549557

**Pull request closed without merging**